File: Project/src/app/models/plane.ts
import { Segment } from './segment';
import { Airline } from './airline.model';

export class Plane {
    id:number;
    code:string;
    segments:Segment[];
    booked:string[]=[];
    airlineId:number;
    airline:Airline;
    airlineName:string;

    constructor(code:string, segments:Segment[], airlineId:number){
        this.code=code;
        this.segments=segments;
        this.airlineId=airlineId;
    }
}

File: Server/Server/Models/Plane.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Server.Models
{
    public class Plane
    {
        [Key]
        public string Code { get; set; }
        public int AirlineId { get; set; }
        [ForeignKey("AirlineId")]
        public virtual Airline Airline { get; set; }
        public ICollection<Segment> Segments { get; set; }
        public ICollection<OccupiedDate> OccupiedDates{ get; set; }
    }
}
